### DISCUSSION TITLE
How to compute plastic strain rate

### DISCUSSION THREAD

**Discussion by RWTHLHK** [idaholab/moose#27611] [Q&A Modules: Solid mechanics]
I need to compute some plastic work requiring a plastic strain rate. However, I looked at the website and the source codes, and currently, there is no way to retrieve the plastic strain rate directly.

**Comment by GiudGiud** [comment]
Use Debug/show_material_props to see if it s computed as a material property
If so, then a MaterialRealAux (or the appropriate type depending on the data type of that property) will do the trick

**Reply by RWTHLHK** [reply]
it seems like plastic strain rate is not computed as a material property. Can (p - p_old)/delta_t work? p is the plastic strain and delta_t is the time increment.

**Reply by GiudGiud** [reply]
If you are using an implicit Euler time integrator, the default, then this will be correct

**Reply by RWTHLHK** [reply]
thanks!
